packages/openmrs/test/index.js

```diff
@@ -47,3 +59,239 @@ describe('execute', () => {
     });
   });
 });
+
+describe('request', () => {
+  it('should search for a patient', async () => {
+    testServer
+      .intercept({
+        path: '/ws/rest/v1/patient',
+        query: { q: 'Sarah 1' },
+        method: 'GET',
+      })
+      .reply(200, { results: [{ display: 'Sarah 1' }] }, { ...jsonHeaders });
+    const state = { configuration };
+
+    const { body } = await request(
+      state,
+      'GET',
+      '/ws/rest/v1/patient',
+      {},
+      { q: 'Sarah 1' }
+    );
+
+    expect(body.results[0].display).to.eql('Sarah 1');
+  });
+  it('should auto-fetch patients with a limit', async () => {
+    testServer
+      .intercept({
+        path: '/ws/rest/v1/patient',
+        query: { q: 'Sarah', limit: 1 },
+        method: 'GET',
+      })
+      .reply(
+        200,
+        {
+          results: [{ display: 'Sarah 1' }],
+          links: [
+            {
+              rel: 'next',
+              uri: 'https://fn.openmrs.org/ws/rest/v1/patient?q=Sarah&limit=1&startIndex=1',
+              resourceAlias: null,
+            },
+          ],
+        },
+        { ...jsonHeaders }
+      );
+    testServer
+      .intercept({
+        path: '/ws/rest/v1/patient',
+        query: { q: 'Sarah', limit: 1, startIndex: 1 },
+        method: 'GET',
+      })
+      .reply(
+        200,
+        {
+          results: [{ display: 'Sarah 2' }],
+        },
+        { ...jsonHeaders }
+      );
+
+    const state = { configuration };
+    const { body } = await request(
+      state,
+      'GET',
+      '/ws/rest/v1/patient',
+      {},
+      { q: 'Sarah', limit: 1 }
+    );
+    expect(body.results[0].display).to.eql('Sarah 1');
+    expect(body.results[1].display).to.eql('Sarah 2');
+  });
+
+  it('should not auto-fetch if the user sets startIdex', async () => {
+    testServer
+      .intercept({
+        path: '/ws/rest/v1/patient',
+        query: { q: 'Sarah', limit: 1, startIndex: 1 },
+        method: 'GET',
+      })
+      .reply(
+        200,
+        {
+          results: [{ display: 'Sarah 2' }],
+        },
+        { ...jsonHeaders }
+      );
+
+    const state = { configuration };
+    const { body } = await request(
+      state,
+      'GET',
+      '/ws/rest/v1/patient',
+      {},
+      { q: 'Sarah', limit: 1, startIndex: 1 }
+    );
+
+    expect(body.results[0].display).to.eql('Sarah 2');
+  });
+});
+
+describe('get', () => {
```

Comment:
Oh, excellent! You didn't need to add tests for the all the operations, I didn't ask you to do that.

But it's great that we have them now - it'll help prevent regressions in future. Thank you!

Reply:
Yeah just very basic test to set the ground if we need to expand more